### PR TITLE
Add support to Windows

### PR DESCRIPTION
**Pull request by Brijeshthummar02**
issue #3 
updated to issue #20

### CODE REVIEW

**Comment by Brijeshthummar02** [comment]
@Cereal84 i request you to go through the file and say changes if needed, thanks.

**Comment by Cereal84** [comment]
@Brijeshthummar02 : Please resolve the conflicts in install.sh file.

**Comment by Brijeshthummar02** [comment]
@Cereal84 i'm not getting what is the conflict.

**Comment by Cereal84** [comment]
If you see this page you can see "This branch has conflicts that must be resolved".
It reports that there is a conflict/s in _install.sh_ file between your branch and the main, these conflicts depends on the fact that the file was updated in _main_ branch and its history goes forward respect your branch.
You can see it [here](https://github.com/Cereal84/AnadiCSV/pull/33/conflicts).

I suggest you to rebase your branch with the _main_ one and resolve locally the conflicts, check this [doc](https://git-scm.com/docs/git-rebase).

```shell
    $ git checkout main
    $ git fetch
    $ git pull origin main
    $ git checkout Install.sh-new
    $ git rebase main
```

If you are not confident on rebase your branch you can update the _main_ branch in your machine and create a new branch based on that _main_, add your changes and push a new PR.

```shell
    $ git checkout main
    $ git fetch
    $ git pull origin main
    $ git checkout -b fix_issue_20
```

File: install.sh [hunk 1]
```diff
@@ -1,9 +1,68 @@
...
+  msys*|cygwin*|mingw*)  
+    OS="Windows"
+    install_windows  # Calling the install function for Windows systems
+    exit 1
+    ;; 
+  *)        
+    echo "Error: Unknown OS: $OSTYPE"
+    exit 1
+    ;;
+esac
```
Comment (Cereal84) [inline install.sh]: For Linux and Windows we just copy anadi.sh file but do not build the image.

I suggest to move the function calling outside the switch case after the build_image.

Are you sure build_image function works on Windows  too ?

Comment (Cereal84) [reply install.sh]: the steps which I imagine in the process :

 - detect the OS
 - check_requirements
     - here we set ENGINE, are you sure that those commands work on windows too?
     
 - build the image based on current OS
     - the bash commands used to create the image work on Windows too?
     
 - build the config path if not exists
     - does Windows support mkdir -p ?
     - have Windows the path "${HOME}/.config/anadi/" ?
     
 - use OS value to call the correct function to install anadi.sh file

**Comment by Brijeshthummar02** [comment]
@Cereal84 now is it fine?

File: install.sh [hunk 2]
```diff
@@ -4,31 +4,72 @@
...
+    echo "Installed anadi.sh to /usr/local/bin/anadi"
+    ;;
+esac
 
-echo "Now you can use the command 'anadi'"
+# Build Docker image after installation (if not handled above)
+./build_image.sh || {
+    echo "Error: Failed to build Docker image"
+    exit 1
+}
```
Comment (Cereal84) [inline install.sh:75]: Build_image.sh file does not exists, please remove these lines

File: install.sh [hunk 3]
```diff
@@ -4,31 +4,72 @@
...
+case "$OS" in
+  "Linux")
+    install_linux
+    ;;
+  "Windows")
+    install_windows
+    ;;
+  "OSX")
+    cp anadi.sh /usr/local/bin/anadi
+    echo "Installed anadi.sh to /usr/local/bin/anadi"
```
Comment (Cereal84) [inline install.sh:67]: Why didn't you add a osx_install function too like Linux and Windows?

File: install.sh [hunk 4]
```diff
@@ -4,31 +4,72 @@
...
+    exit 1
+    ;;
 esac
 
-echo -e "OS: ${Green}${OS}${NC}" 
+# Notify about the detected OS
+echo -e "OS: ${Green}${OS}${NC}"
 
+# Build image (this assumes the build_image function is defined in utils.sh)
 build_image $OS $ENGINE
```
Comment (Cereal84) [inline install.sh:35]: Are we sure that that function work on Windows too?
I mean, the commands inside works for Windows too?

File: install.sh [hunk 5]
```diff
@@ -4,31 +4,72 @@
...
 # --- Variables
-
 CONF_PATH="${HOME}/.config/anadi/"
 BIN_DESTINATION_PATH="$HOME"
 OS=""
 
-
-# makes some checks and set ENGINE variable
+# Makes some checks and sets ENGINE variable
 check_requirements
```
Comment (Cereal84) [inline install.sh:12]: Does the  way we check for docker or podman works on Windows too?

**Comment by Brijeshthummar02** [comment]
i'm creating new pr due to some reasons and closing this one.

**Pull request closed without merging**